packages/sdk/src/jwt/core/validate.ts

```diff
@@ -62,8 +68,13 @@ export function verifyJWT(jwt: string, expectedAudience: string): VincentJWT {
     const r = signatureBytes.slice(0, 32);
     const s = signatureBytes.slice(32, 64);
 
+    const { app, authentication, pkp } = decoded.payload;
+    if (!isDefinedObject(app) || !isDefinedObject(authentication) || !isDefinedObject(pkp)) {
+      throw new Error(`Missing required fields in JWT payload`);
+    }
```

Comment:
1. For consistency with the rest of our error syntax, we should still prefix this w/ the appropriate `JWT_ERROR` string.
2. Also, for the benefit of someone debugging any issue and for us to identify what's wrong if we get a report of us providing invalid JWTs to consumers, it would be good to be verbose about what was missing that we expected to be there.

```suggestion
    const { app, authentication, pkp } = decoded.payload;

    if (!isDefinedObject(app)) {
      throw new Error(`${JWT_ERROR.INVALID_JWT}: Missing "app" field in JWT payload.`);
    }

    if (!isDefinedObject(authentication)) {
      throw new Error(`${JWT_ERROR.INVALID_JWT}: Missing "authentication" field in JWT payload.`);
    }

    if (!isDefinedObject(pkp)) {
      throw new Error(`${JWT_ERROR.INVALID_JWT}: Missing "pkp" field in JWT payload.`);
    }
```

3. This check should happen outside of this `try{}` block, above this where we do the other verifications, so that we don't convert these errors into `${JWT_ERROR.INVALID_SIGNATURE}: Invalid signature: ${(error as Error).message}` on L103 below even though missing properties on the JWT isn't a bad signature issue.

Reply:
Good points